### PR TITLE
fix: Check the value of rt_sink_connected_to is a dict

### PR DESCRIPTION
**Pull request by abtreece**
### What does this PR do?

Added a condition to ensure the returned value of `stats['sinks']['sink_stats']['rt_sink_connected_to']` is a dict before we gather those items.

### Motivation

Riak replication is frequently "too busy" to report the realtime sink metrics. This results in errors from the check...

```
      Error: 'str' object has no attribute 'items'
      Traceback (most recent call last):
        File "/opt/datadog-agent/embedded/lib/python3.8/site-packages/datadog_checks/base/checks/base.py", line 841, in run
          self.check(instance)
        File "/opt/datadog-agent/embedded/lib/python3.8/site-packages/datadog_checks/riak_repl/riak_repl.py", line 125, in check
          for key, val in iteritems(stats['sinks']['sink_stats']['rt_sink_connected_to']):
        File "/opt/datadog-agent/embedded/lib/python3.8/site-packages/six.py", line 589, in iteritems
          return iter(d.items(**kw))
      AttributeError: 'str' object has no attribute 'items'
```

### Review checklist

- [X] PR has a [meaningful title](https://github.com/DataDog/integrations-core/blob/master/CONTRIBUTING.md#pull-request-title) or PR has the `no-changelog` label attached
- [X] Feature or bugfix has tests
- [X] Git history is clean
- [ ] If PR impacts documentation, docs team has been notified or an issue has been opened on the [documentation repo](https://github.com/DataDog/documentation/issues/new)

### CODE REVIEW

**Review by ruthnaebeck — APPROVED** [review]
👍  for docs

**Comment by hithwen** [comment]
Hi, the tests on this PR are broken because of an error on our setup. Once this is merged, if you update your branch the build should be able to run fine https://github.com/DataDog/integrations-extras/pull/1088#pullrequestreview-815828267

File: riak_repl/manifest.json [hunk 1]
```diff
@@ -2,7 +2,7 @@
   "integration_id": "riak-repl",
   "display_name": "Riak MDC Replication",
   "maintainer": "britt.treece@gmail.com",
-  "manifest_version": "1.0.0",
+  "manifest_version": "1.0.1",
```
Comment (sarah-witt) [inline riak_repl/manifest.json:5]:
```suggestion
  "manifest_version": "1.0.0",
```
This does not need to be updated

File: riak_repl/CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,9 @@
 # CHANGELOG - Riak_Repl
 
+## 1.0.1
+
+ * [FIX] add a condition to check that `rt_sink_connected_to` is a dict before collecting `sink_stats`
```
Comment (sarah-witt) [inline riak_repl/CHANGELOG.md:5]:
```suggestion
 * [FIX] Add a condition to check that `rt_sink_connected_to` is a dict before collecting `sink_stats`
```
A small nit to update capitalization

**Review by sarah-witt — CHANGES_REQUESTED** [review]
Hi @abtreece, your changes looks good, I just had two comments!

File: riak_repl/manifest.json (hunk 1 above)
Comment (abtreece) [reply riak_repl/manifest.json:5]: The manifest version and the code version aren't the same thing?

File: riak_repl/CHANGELOG.md (hunk 2 above)
Comment (abtreece) [reply riak_repl/CHANGELOG.md:5]: If I cap line 5 I'll be compelled to fix the other lines so they are congruent... thoughts?

File: riak_repl/manifest.json (hunk 1 above)
Comment (sarah-witt) [reply riak_repl/manifest.json:5]: No, it is referring to the version of style of manifest used, as described here: https://docs.datadoghq.com/developers/integrations/check_references/#manifest-file. The only values we support for this field are 1.0.0 and 2.0.0

File: riak_repl/CHANGELOG.md (hunk 2 above)
Comment (sarah-witt) [reply riak_repl/CHANGELOG.md:5]: Up to you, either way is fine

**Review by sarah-witt — COMMENTED** [review]
Looks great! One more thing- It looks like style is failing on your PR. If you run `ddev test riak_repl -fs` It will fix the errors and/or instruct you on what is wrong.

**Review by sarah-witt — APPROVED** [review]
Looks good to me, thanks!

**Pull request merged**